lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -49,15 +49,20 @@ export class PublicClientApplication extends ClientApplication implements IPubli
         this.initializeBrokering();
     }
 
+    /**
+     * 
+     */
     private initializeBrokering(): void {
         if (this.config.system.brokerOptions.actAsBroker) {
-            this.broker = new BrokerManager(this.config.system.brokerOptions, this.logger, version);
-            this.logger.verbose("Acting as Broker");
-            this.broker.listenForMessage();
+            this.broker = new BrokerClientApplication(this.config);
         } else if (this.config.system.brokerOptions.allowBrokering) {
-            this.embeddedApp = new BrokerClient(this.config.system.brokerOptions, this.logger, this.config.auth.clientId,  version);
+            this.embeddedApp = new EmbeddedClientApplication(this.config, this.logger, version, this.browserStorage);
```

Comment:
One more nitpick: `version` can just be imported directly in `EmbeddedClientApplication`

Reply:
done.